Formula/python.rb

```diff
@@ -74,15 +74,23 @@ def install
     ldflags  = []
     cppflags = []
 
+    # Need this to use homebrew installed libraries.
+    if HOMEBREW_PREFIX.to_s != "/usr/local"
+      cppflags << "-I#{HOMEBREW_PREFIX}/include"
```

Comment:
Instead you should add multiple `-I`and `-L` for the each of the libraries that are required rather than the whole prefix.

Reply:
It may be more than that. I believe (not confirmed) that these paths are stored in the final configuration of the Python interpreter. When building Python packages that include native code, we want them to link against brewed libraries, or it won't be able to find the necessary dependent libraries.

Reply:
If that's the case we need to figure out another solution. This is a sledgehammer we explicitly avoid doing due to the opportunistic linking that results.

Reply:
If I'm right, and that information is stored in a text, it could be patched after building Python.

Reply:
> This is a sledgehammer we explicitly avoid doing due to the opportunistic linking that results.

Right, that was my concern too.

Also, shouldn't homebrew's superenv be making explicit dependencies available on the include and library search path, no matter where the homebrew prefix is?  

@m000 Do you know which libraries this affects for you?

Reply:
From the original comments, it sounds like this is only for `gdbm` so perhaps there's an issue with how `gdbm` is getting installed with a non-standard prefix, or something in the superenv that needs to be looked at. Hopefully @m000 can confirm or deny this assertion.

Reply:
> Also, shouldn't homebrew's superenv be making explicit dependencies available on the include and library search path, no matter where the homebrew prefix is?

Reply:
> Also, shouldn't homebrew's superenv be making explicit dependencies available on the include and library search path, no matter where the homebrew prefix is?

It's not a problem with Superenv. Python is parsing the `-I` options to use those values for other purposes. We had the same issue in Linuxbrew. See this comment and corresponding fix in Linuxbrew-core.

```
    # Python's setup.py parses CPPFLAGS and LDFLAGS to learn search
    # paths for the dependencies of the compiled extension modules.
    # See Linuxbrew/linuxbrew#420, Linuxbrew/linuxbrew#460, and Linuxbrew/linuxbrew#875
```
https://github.com/Homebrew/linuxbrew-core/blob/77223c8d564be4611c518046f71da06df0871326/Formula/python.rb#L118-L129

Reply:
So I think we want to fix this after the fact, no? i.e. express explicit build time dependencies but then patch python once it's built? I'm a bit hazy on how we should proceed here. I agree with Mike that setting 	`-I` and `-L` in the formula should be avoided. The proposed change here is effectively to adopt linuxbrew's approach.

Reply:
That makes sense to me, to use specific `-I` options during build, and then patch after the build is complete. You could add `-Ifoobar` and then search through the result files to find in which file `foobar` is stored.

Reply:
Or what about something like `@HOMEBREW_PREFIX@` so you also know what is supposed to be there, and it won't get clobbered by projects doing stuff with foobar...

Reply:
I'd suggest avoiding the exact string `@HOMEBREW_FIX@`, but yes, you could use something like `-IXYZZY` that you could then search and replace afterward. This could potentially be in the formula. I was thinking just as a reconnaissance mission (outside of the formula), to learn in which file this information is stored.